### PR TITLE
[New Extension] Sidekick

### PR DESCRIPTION
**Pull request by shikibu9419**
## Screencast

sidekick-1

sidekick-2

sidekick-3

sidekick-4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/sidekick/package.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      "description": "Open a new Sidekick tab.",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "useOriginalFavicon",
+      "title": "Fav icon",
+      "label": "Extract the favicon from every open tab",
+      "description": "Before enable this, you should enable 'View > Developer > Allow JavaScript from Apple Events' in every Sidekick profile.",
```
Comment (thomaslombart) [inline extensions/sidekick/package.json:47]:
```suggestion
      "description": "Before enabling this, you should enable 'View > Developer > Allow JavaScript from Apple Events' in every Sidekick profile.",
```

File: extensions/sidekick/src/browser/history.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
+import path from "path";
```
Comment (thomaslombart) [inline extensions/sidekick/src/browser/history.ts:1]: Note that you can use `useSQL` hook from the `@raycast/utils` package to make SQL stuff easier to manage: https://developers.raycast.com/utilities/react-hooks/usesql 😊

**Review by thomaslombart — CHANGES_REQUESTED** [review]
Thanks for the extension! 🚀

As I'm not a Sidekick user, I got some errors related to empty states when testing the extension.

If I don't have any bookmarks installed or if the app isn't installed, I have the following error screen:

CleanShot 2023-02-08 at 10 41 46@2x

I guess this happens because there are no bookmarks in the system if no one is created. You could typically handle this using `List.EmptyView` and an action to invite the user to add some bookmarks if the `Bookmarks` folder doesn't exist.

Could you also add a description to the PR to briefly introduce the extension? Thanks!

**Comment by shikibu9419** [comment]
Oh, thank you for your reviews!!
I'll fix it; please wait.

**Pull request closed without merging**